host_core/test/e2e/kvcounter_test.exs

```diff
@@ -88,4 +92,77 @@ defmodule HostCore.E2E.KVCounterTest do
     {:ok, resp} = HTTPoison.get("http://localhost:8081/foobar")
     assert resp.body == "{\"counter\":#{incr_count}}"
   end
+
+  test "kvcounter unprivileged access denied" do
+    {:ok, bytes} = File.read(@kvcounter_unpriv_path)
+    Logger.info("attempting to load kvcounter actor")
+    Logger.info("Here's the lattice prefix btw: #{HostCore.Host.lattice_prefix()}")
+    {:ok, _pid} = HostCore.Actors.ActorSupervisor.start_actor(bytes)
+    Logger.info("successfully started kvcounter actor")
+    on_exit(fn -> HostCore.Actors.ActorSupervisor.terminate_actor(@kvcounter_key, 1) end)
+
+    {:ok, _pid} =
+      HostCore.Providers.ProviderSupervisor.start_executable_provider(
+        @httpserver_path,
+        @httpserver_key,
+        @httpserver_link,
+        @httpserver_contract
+      )
+
+    on_exit(fn ->
+      HostCore.Providers.ProviderSupervisor.terminate_provider(@httpserver_key, @httpserver_link)
+    end)
+
+    {:ok, _pid} =
+      HostCore.Providers.ProviderSupervisor.start_executable_provider(
+        @redis_path,
+        @redis_key,
+        @redis_link,
+        @redis_contract
+      )
+
+    on_exit(fn ->
+      HostCore.Providers.ProviderSupervisor.terminate_provider(@redis_key, @redis_link)
+    end)
+
+    actor_count =
+      Map.get(HostCore.Actors.ActorSupervisor.all_actors(), @kvcounter_unpriv_key)
+      |> length
+
+    assert actor_count == 1
+
+    assert HostCore.Providers.ProviderSupervisor.all_providers() == [
+             {@httpserver_key, @httpserver_link, @httpserver_contract},
+             {@redis_key, @redis_link, @redis_contract}
+           ]
+
+    Process.sleep(5000)
```

Comment:
Did you find that you needed a 5s sleep or else you'd get failures?

Reply:
this was just copied from the other test. I can probably drop this delay.